Sources/PointFreeRouter/Util.swift

```diff
@@ -0,0 +1,99 @@
+import ApplicativeRouter
+import HttpPipeline
+import Models
+import Prelude
+import Tagged
+import UrlFormEncoding
+
+func payload<A, B>(
+  _ iso1: PartialIso<String, A>,
+  _ iso2: PartialIso<String, B>,
+  separator: String = "--POINT-FREE-BOUNDARY--"
+  )
+  -> PartialIso<String, (A, B)> {
+
+    return PartialIso<String, (A, B)>(
+      apply: { payload in
+        let parts = payload.components(separatedBy: separator)
+        guard
+          let first = parts.first.flatMap(iso1.apply),
+          let second = parts.last.flatMap(iso2.apply) else { return nil }
+        return (first, second)
+    },
+      unapply: { first, second in
+        guard
+          let first = iso1.unapply(first),
+          let second = iso2.unapply(second)
+          else { return nil }
+        return "\(first)\(separator)\(second)"
+    })
+}
+
+extension PartialIso where A == String, B == String {
+  static func decrypted(withSecret secret: String) -> PartialIso<String, String> {
+    return PartialIso(
+      apply: { HttpPipeline.decrypted(text: $0, secret: secret) },
+      unapply: { encrypted(text: $0, secret: secret) }
```

Comment:
one bummer is that I had to bring in `HttpPipeline` to get access to encryption. maybe our `swift-partial-iso` package could have a sub package for encryption isos